BrowserKit/Sources/Common/Extensions/URLExtension.swift

```diff
@@ -0,0 +1,242 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+
+extension URL {
+    /**
+     * Returns a shorter displayable string for a domain
+     *
+     * E.g., https://m.foo.com/bar/baz?noo=abc#123  => foo
+     *       https://accounts.foo.com/bar/baz?noo=abc#123  => accounts.foo
+     **/
```

Comment:
nit, if we move the code we could use standard comment (as given through option+cmd+/ shortcut with `///`). There's a couple of places in the PR the comment applies.